frontendVue3/src/components/upload_cards/file_create/ActionButtons.vue

```diff
@@ -45,49 +45,82 @@ let loading = ref(false);
       console.log("Single upload");
       uploadSingle();
       } else {
-        console.log("Bulk upload");
-        uploadBulk();
-    } 
+      console.log("Bulk upload");
+      uploadBulk();
     }
   } else {
     emit('submitClick');
   }
 }
-async function uploadSingle(){
-            loading.value = true;
-            const data = new FormData();
-            let file_data = [];
-            let file_object = {};
-            file_object['label'] = props.data.label;
-            file_object['description'] = props.data.description;
-            file_object['license'] = props.data.license;
-            file_object['version'] = props.data.version;
-            file_object['family'] = props.data.family;
-            file_object['tags'] = props.data.tags;
-            file_object['file'] = '0';
-            file_data.push(file_object);
-            data.append('0', props.data.files[0]);
-            data.append('files', JSON.stringify(file_data));
-            uploadStatus.value = 'Uploading file...';
-            await fileStore.uploadBulkFeatureModels(data);
-            uploadInfo.value = {
-              format: "Single",
-              fileNames: props.data.label,
-              license: props.data.license
-            };
-            uploadStatus.value = '';
-            loading.value = false;
-            emit('uploadSuccessfull', uploadInfo.value);
-            console.log("Emit Upload")
-            //emit('close');
+async function uploadSingle() {
+  loading.value = true;
+  const data = new FormData();
+  let file_data = [];
+  let file_object = {};
+  file_object['label'] = props.data.label;
+  console.log(props.data.label)
+  file_object['description'] = props.data.description;
+  file_object['license'] = props.data.license;
+
+
+  file_object['family'] = props.data.family;
+  file_object['tags'] = props.data.tags;
+  file_object['file'] = '0';
+  if (!props.data.files.some(file => file.name.endsWith('.zip'))) {
+    console.log("Upload Single");
+    file_object['version'] = props.data.version;
+    file_data.push(file_object);
+    data.append('0', props.data.files[0]);
+    data.append('files', JSON.stringify(file_data));
+  } else {
+    data.append('files', JSON.stringify(file_object));
+    data.append('file', props.data.files[0]);
+  }
+
+
+  uploadStatus.value = 'Uploading file...';
+  if (props.data.files.some(file => file.name.endsWith('.zip'))) {
+    console.log("Upload ZIP");
+    let success = await fileStore.uploadZipFeatureModels(data)
+      if (success){
+        uploadInfo.value = {
+        format: "ZIP",
+        fileNames: props.data.label,
+        license: props.data.license
+
+      };
+      uploadStatus.value = '';
+      loading.value = false;
+      emit('uploadSuccessfull', uploadInfo.value);
+      }
+      else{
+        handleunsuccessfullupload();
+      }
+  } else {
+     let success = await fileStore.uploadBulkFeatureModels(data)
+      if(success){
+        uploadInfo.value = {
```

Comment:
This code( when upload succesful) is used a few times in the component, maybe refactoring to function  handleSuccessfulUpload could be a idea especially because some value resets take place.